src/components/templates/ProductTemplate/ProductTemplateNavigation/ProductTemplateNavigation.tsx

```diff
@@ -31,16 +31,16 @@ const BackText = styled(Text).attrs({ color: colors.greyDark })`
   line-height: 0;
 `;
 
-export function ProductTemplateNavigation({ prevStep }: ProductTemplateNavigationProps) {
+export function ProductTemplateNavigation({ prevStep, onBackPressed }: ProductTemplateNavigationProps) {
   return (
     <Wrapper className="my-4 m:mt-0" data-automation="ZA.ProductTemplateNavigation">
-      {typeof prevStep === 'string' ? (
-        <StyledLink href={prevStep} aria-label="Back">
+      {typeof prevStep === 'object' ? (
+        prevStep
+      ) : (
+        <StyledLink href={prevStep as string | undefined} aria-label="Back" onClick={onBackPressed}>
```

Comment:
We have to avoid using type casting.

In this case I think it will look better like this:

```ts
{typeof prevStep === 'string' ? (
        <StyledLink href={prevStep} aria-label="Back" onClick={onBackPressed}>
          <Icon variant={faChevronLeft} color={colors.greyDark} size="xs" className="mr-2" />
          <BackText>Back</BackText>
        </StyledLink>
      ) : (
        prevStep
      )}
```

Then you also have to fix the type of `prevStep` because currently, it supports `boolean` and `number` which is wrong. I know it's not your fault and it was already there... but Can you try to make it work like:

```ts
  prevStep?: ReactElement | string;
```
In theory, this is the correct way, but I'm not 100% sure since my memory is not that good :P